tests/pipeline/test_engine.py

```diff
@@ -510,6 +519,107 @@ def test_rolling_and_nonrolling(self):
                 ),
             )
 
+    def test_factor_with_single_output(self):
+        dates = self.dates[5:10]
+        assets = self.assets
+        num_dates = len(dates)
+        open = USEquityPricing.open
+        open_values = array([self.constants[open]] * num_dates, dtype=float)
+        engine = SimplePipelineEngine(
+            lambda column: self.loader, self.dates, self.asset_finder,
+        )
+
+        single_output = OpenPrice(outputs=['open'])
+        pipeline = Pipeline(columns={'open': single_output.open})
+
+        results = engine.run_pipeline(pipeline, dates[0], dates[-1])
+        output_results = results['open'].unstack()
+        output_expected = {asset: open_values for asset in assets}
+
+        assert_frame_equal(
+            output_results,
+            DataFrame(output_expected, index=dates, columns=assets),
+        )
+
+    def test_factor_with_multiple_outputs(self):
+        dates = self.dates[5:10]
+        assets = self.assets
+        asset_ids = self.asset_ids
+        constants = self.constants
+        num_dates = len(dates)
+        num_assets = len(assets)
+        open = USEquityPricing.open
+        close = USEquityPricing.close
+        engine = SimplePipelineEngine(
+            lambda column: self.loader, self.dates, self.asset_finder,
+        )
+
+        def create_expected_results(expected_value, mask):
+            expected_values = where(mask, expected_value, nan)
+            return DataFrame(expected_values, index=dates, columns=assets)
+
+        multiple_outputs = MultipleOutputs()
+        open_price, close_price = MultipleOutputs()
+
+        # Ensure that both methods of accessing our outputs return the same
+        # things.
+        self.assertIs(open_price, multiple_outputs.open)
+        self.assertIs(close_price, multiple_outputs.close)
+
+        pipeline = Pipeline(
+            columns={'open_price': open_price, 'close_price': close_price},
+        )
+
+        results = engine.run_pipeline(pipeline, dates[0], dates[-1])
+        first_output_results = results['open_price'].unstack()
+        second_output_results = results['close_price'].unstack()
+
+        first_output_expected = create_expected_results(
+            constants[open], full((num_dates, num_assets), True),
+        )
+        second_output_expected = create_expected_results(
+            constants[close], full((num_dates, num_assets), True),
+        )
+
+        assert_frame_equal(first_output_results, first_output_expected)
+        assert_frame_equal(second_output_results, second_output_expected)
+
+        # Now test the same outputs again, but with different masks.
```

Comment:
One option here might be to parameterize the entire suite on mask with values of `alternating_mask, cascading_mask, NotSpecified`, and then just always pass mask to the factor constructors.  This makes the logic in `create_expected_results` slightly more complex, but it would cut the size of the test in half.